### PR TITLE
add distributed Cmdlet execution service

### PR DESCRIPTION
**Pull request by huafengw**
This is the initial pull request, which mainly focus on the framework design, and simple cross-jvm Cmdlet execution is already verified.

### CODE REVIEW

File: smart-actions/src/main/java/org/smartdata/actions/PrintAction.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.smartdata.actions;
+
+public class PrintAction extends SmartAction {
```
Comment (drankye) [inline smart-actions/src/main/java/org/smartdata/actions/PrintAction.java]: Cool! Maybe name it as HelloAction?

File: smart-server/src/main/java/org/smartdata/server/cmdlet/CmdletManager.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  public void stop() throws IOException {
+    this.executorService.shutdown();
+  }
+
+  @Override
+  public void join() throws IOException {
+
+  }
+
+  public long submitCommand(String command) throws IOException {
```
Comment (drankye) [inline smart-server/src/main/java/org/smartdata/server/cmdlet/CmdletManager.java]: Command -> Cmdlet?

File: smart-server/src/main/java/org/smartdata/server/cmdlet/executor/CmdletStatusReporter.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.smartdata.server.cmdlet.executor;
+
+public interface CmdletStatusReporter {
+  void report(Object status);
```
Comment (drankye) [inline smart-server/src/main/java/org/smartdata/server/cmdlet/executor/CmdletStatusReporter.java]: Maybe: void report(CmdletStatus status)

Comment (huafengw) [inline smart-server/src/main/java/org/smartdata/server/cmdlet/executor/CmdletStatusReporter.java]: I will add this interface in follow-on pr.

**Pull request merged**